src/rpc/governance.cpp

```diff
@@ -199,9 +202,23 @@ UniValue gobject_prepare(const JSONRPCRequest& request)
 
     EnsureWalletIsUnlocked();
 
+    // If specified, spend this outpoint as the fee
+    COutPoint outpoint;
+    outpoint.SetNull();
+    if (request.params.size() == 8) {
+        uint256 collateralHash = ParseHashV(request.params[6], "outputHash");
+        int32_t collateralIndex = ParseInt32V(request.params[7], "outputIndex");
+        if (collateralHash.IsNull() || collateralIndex < 0) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("invalid hash or index: %s-%d", collateralHash.ToString(), collateralIndex));
+        }
+        outpoint = COutPoint(collateralHash, (uint32_t)collateralIndex);
+    }
+
     CWalletTx wtx;
-    if (!pwalletMain->GetBudgetSystemCollateralTX(wtx, govobj.GetHash(), govobj.GetMinCollateralFee(), useIS)) {
-        throw JSONRPCError(RPC_INTERNAL_ERROR, "Error making collateral transaction for governance object. Please check your wallet balance and make sure your wallet is unlocked.");
+    if (!pwalletMain->GetBudgetSystemCollateralTX(wtx, govobj.GetHash(), govobj.GetMinCollateralFee(), useIS, outpoint)) {
+        std::string err = "Error making collateral transaction for governance object. Please check your wallet balance and make sure your wallet is unlocked.";
+        if (request.params.size() == 7) err += "Please verify your specified output is valid and is enough for the combined proposal fee and transaction fee.";
```

Comment:
This will never be used because `param.size()` can't be 7 due to the conditions above.
```suggestion
```

Reply:
I prefer https://github.com/dashpay/dash/pull/2482/commits/c510880e7584a64a453e9b392b90ee7937423abf